source/common/event/file_event_impl.h

```diff
@@ -18,6 +18,7 @@ class FileEventImpl : public FileEvent, ImplBase {
 public:
   FileEventImpl(DispatcherImpl& dispatcher, os_fd_t fd, FileReadyCb cb, FileTriggerType trigger,
                 uint32_t events);
+  ~FileEventImpl() override { activation_cb_->cancel(); }
```

Comment:
This shouldn't be needed.  The deletion of the activation_cb_ implicitly cancels the callback.

Reply:
love RAII ! Now that this is not the culprit we can restrict this PR to user space only

Reply:
This part is reverted